pgo/cmd/pgadmin.go

```diff
@@ -0,0 +1,219 @@
+package cmd
+
+/*
+ Copyright 2018 - 2020 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	msgs "github.com/crunchydata/postgres-operator/apiservermsgs"
+	"github.com/crunchydata/postgres-operator/pgo/api"
+	"github.com/crunchydata/postgres-operator/pgo/util"
+)
+
+// showPgAdminTextPadding contains the values for what the text padding should be
+type showPgAdminTextPadding struct {
+	ClusterName int
+	ClusterIP   int
+	ExternalIP  int
+	ServiceName int
+}
+
+// updatePgAdminTextPadding contains the values for what the text padding should be
+type updatePgAdminTextPadding struct {
+	ClusterName  int
+	ErrorMessage int
+	Status       int
+}
+
+func createPgAdmin(args []string, ns string) {
+	if Selector == "" && len(args) == 0 {
+		fmt.Println("Error: The --selector flag is required when cluster is unspecified.")
+		return
+	}
+
+	request := msgs.CreatePgAdminRequest{
+		Args:          args,
+		ClientVersion: msgs.PGO_VERSION,
+		Namespace:     ns,
+		Selector:      Selector,
+	}
+
+	response, err := api.CreatePgAdmin(httpclient, &SessionCredentials, &request)
+
+	if err != nil {
+		fmt.Println("Error: " + err.Error())
+		os.Exit(1)
+	}
+
+	// this is slightly rewritten from the legacy method
+	if response.Status.Code != msgs.Ok {
+		fmt.Println("Error: " + response.Status.Msg)
+		for _, v := range response.Results {
+			fmt.Println(v)
+		}
+		os.Exit(1)
+	}
+
+	for _, v := range response.Results {
+		fmt.Println(v)
+	}
+}
+
+func deletePgAdmin(args []string, ns string) {
+	if Selector == "" && len(args) == 0 {
+		fmt.Println("Error: The --selector flag or a cluster name is required.")
+		return
+	}
+
+	// set up the API request
+	request := msgs.DeletePgAdminRequest{
+		Args:          args,
+		ClientVersion: msgs.PGO_VERSION,
+		Selector:      Selector,
+		Namespace:     ns,
+	}
+
+	response, err := api.DeletePgAdmin(httpclient, &SessionCredentials, &request)
+	if err != nil {
+		fmt.Println("Error: " + err.Error())
+		os.Exit(2)
```

Comment:
`2` is fine, but does it mean something different than during `create` which uses `1`?

Reply:
We have `2` throughout the code, where AIUI we should be using `1`, so it will be eventually consistent.